src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const returnedArray = [];
+  const cloneState = { ...state };
+  // returnedArray.push( {...state });
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(cloneState, obj.extraData);
+    } else if (obj.type === 'removeProperties') {
+      for (const i of obj.keysToRemove) {
```

Comment:
```suggestion
      for (const key of obj.keysToRemove) {
```
Make the variable names more informative (also think of a better name for `returnedArray`)